datalad/config.py

```diff
@@ -74,12 +80,35 @@ def get_git_version(runner=None):
                       protocol=StdOutErrCapture)['stdout'].split()[2]
 
 
-def _where_reload(obj):
+def _scope_reload(obj):
     """Helper decorator to simplify providing repetitive docstring"""
-    obj.__doc__ = obj.__doc__ % _where_reload_doc
+    obj.__doc__ = obj.__doc__ % _scope_reload_doc
     return obj
 
 
+#
+# TODO: remove in/after 0.16.0
+#
+def _where_to_scope(func):
+    @wraps(func)
+    def wrapper(*args, **kwargs):
+        if 'where' in kwargs:
+            if 'scope' in kwargs:
+                raise ValueError("Do not specify both 'scope' and DEPRECATED 'wraps'")
+            kwargs = kwargs.copy()
+            where = kwargs.pop('where')
+            if where == 'dataset':
+                warnings.warn("'where=\"dataset\"' is being deprecated as of 0.15. Use 'scope=\"branch\"' instead",
```

Comment:
```suggestion
                warnings.warn("'where=\"dataset\"' is deprecated, use 'scope=\"branch\"' instead",
```

I think we stick with simple statements.

Reply:
BTW, touches on the topic where a better formalized support for deprecations would help -- it is useful to know (without discovery via commits etc analysis) when did we deprecated the feature, so when you encounter such a warning and know current version you can right away assess that may be it is time to remove it (and if you don't know the version when deprecated -- you would just skip/postpone/ignore since might have been in the same "major" release)